### PR TITLE
Decouple mgr push

### LINKED ISSUE #4725: mgrpush gets removed on salt minion - Uyuni 2021.12
mgrpush needs to be installed on client so that RPMs can be pushed to uyuni channels.

Install mgrpush onto salt minion
```
sudo zypper in rhnpush
The following 4 NEW packages are going to be installed:
  mgr-push python3-mgr-push python3-spacewalk-client-tools spacewalk-client-tools
```

Then packages can be pushed to uyuni server
` mgrpush -v --ca-chain=/etc/pki/trust/anchors/RHN-ORG-TRUSTED-SSL-CERT --server=<server> -c <channel> <package>.rpm`

The next time the highstate is run, this package gets removed

```
          ID: remove_traditional_stack
    Function: pkg.removed
      Result: True
     Comment: The following packages were not installed: mgr-cfg, rhncfg, suseRegisterInfo The following packages were removed: spacewalk-client-tools.
     Started: 11:19:50.484621
    Duration: 7275.943 ms
     Changes:   
              ----------
              mgr-push:
                  ----------
                  new:
                  old:
                      4.3.1-2.1.uyuni
              python3-mgr-push:
                  ----------
                  new:
                  old:
                      4.3.1-2.1.uyuni
              python3-spacewalk-client-tools:
                  ----------
                  new:
                  old:
                      4.3.4-2.2.uyuni
              spacewalk-client-tools:
                  ----------
                  new:
                  old:
                      4.3.4-2.2.uyuni
```

### PR DESCRIPTION
**Pull request by ycedres**
## What does this PR change?

`mgr-push` has a dependency on `spacewalk-client-tools`. This causes `mgr-push` to be removed when the traditional stack is removed.

This PR causes `mgr-push` to no longer depend on the traditional stack.

## GUI diff

No difference.

- [X] **DONE**

## Documentation
- No documentation needed: only internal and user invisible changes

- [X] **DONE**

## Test coverage
- ~~No tests: A PR with new unit tests for mgr-push is currently under review and will hopefully be merged soon. 
https://github.com/uyuni-project/uyuni/pull/5654~~. That PR was already merged.
- No tests: already covered. 

- [X] **DONE**

## Links

Fixes #4725 
Tracks https://github.com/SUSE/spacewalk/issues/17775

- [X] **DONE**

## Changelogs

Make sure the changelogs entries you are adding are compliant with https://github.com/uyuni-project/uyuni/wiki/Contributing#changelogs and https://github.com/uyuni-project/uyuni/wiki/Contributing#uyuni-projectuyuni-repository

If you don't need a changelog check, please mark this checkbox:

- [ ] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

### CODE REVIEW

**Review by agraul — COMMENTED** [review]
It would have helped with the PR review to have a bit more context in the git commits and the PR description. I think I now understand how the changes you have done remove the dependency, but it could have been a bit quicker with more information up front.

Based on your comment about the tests being added in a different PR, I assume this PR is blocked by it? Can we run the (new) tests on this PR to check for regressions after merging the other PR?

**Comment by ycedres** [comment]
Thank you @agraul. Sorry if there was a lack of context in the PR description. Yes, as you point out, I would prefer the other PR to be merged first and then run the tests in this PR before merging it. Those tests already cover some functionality that could be affected by changes in this PR.

**Comment by ycedres** [comment]
@agraul now the changes in this PR are covered by the tests triggered in the checks.

**Review by agraul — APPROVED** [review]
In case you've been waiting for me to add a green checkmark, here it is :)

I would suggest to squash the "update changelog" commit into the main commit after rebasing the PR. Having an "update changelog" commit in `git log` is not really useful.

**Pull request closed without merging**
